delfin/drivers/ibm/ds8k/ds8k.py

```diff
@@ -0,0 +1,174 @@
+                if system.get('capalloc') != '' and \
+                        system.get('capalloc') is not None:
+                    used = int(system.get('capalloc'))
+                if system.get('capavail') != '' and \
+                        system.get('capavail') is not None:
+                    free = int(system.get('capavail'))
+                result = {
+                    'name': name,
+                    'vendor': 'IBM',
+                    'model': model,
+                    'status': status,
+                    'serial_number': serial_number,
+                    'firmware_version': version,
+                    'location': '',
+                    'total_capacity': total,
+                    'raw_capacity': raw,
+                    'used_capacity': used,
+                    'free_capacity': free
+                }
+                break
+        return result
+
+    def list_storage_pools(self, context):
+        pool_info = self.rest_handler.get_all_pools()
+        pool_list = []
+        status = constants.StoragePoolStatus.NORMAL
+        if pool_info is not None:
+            pool_data = pool_info.get('data', {}).get('pools')
+            for pool in pool_data:
+                if pool.get('stgtype') == 'fb':
+                    pool_type = constants.StorageType.BLOCK
+                else:
+                    pool_type = constants.StorageType.FILE
+                if (int(pool.get('capalloc')) / int(pool.get('cap'))) * 100 > \
+                        int(pool.get('threshold')):
+                    status = constants.StoragePoolStatus.ABNORMAL
+                pool_result = {
+                    'name': pool.get('name'),
+                    'storage_id': self.storage_id,
+                    'native_storage_pool_id': str(pool.get('id')),
+                    'status': status,
+                    'storage_type': pool_type,
+                    'total_capacity': int(pool.get('cap')),
+                    'used_capacity': int(pool.get('capalloc')),
+                    'free_capacity': int(pool.get('capavail'))
+                }
+                pool_list.append(pool_result)
+        return pool_list
+
+    def list_volumes(self, context):
+        volume_list = []
+        pool_list = self.rest_handler.get_all_pools()
+        if pool_list is not None:
+            pool_data = pool_list.get('data', {}).get('pools')
+            for pool in pool_data:
+                volumes = self.rest_handler.get_pool_volumes(pool.get('id'))
+                if volumes is not None:
+                    vol_entries = volumes.get('data', {}).get('volumes')
+                    for volume in vol_entries:
+                        total = volume.get('cap')
+                        used = volume.get('capalloc')
+                        vol_type = constants.VolumeType.THIN
+                        if volume.get('stgtype') == 'fb':
+                            vol_type = constants.VolumeType.THICK
```

Comment:
Suggest to use `vol_type = constants.VolumeType.THIN if volume.get('stgtype') == 'fb' else constants.VolumeType.THICK` , and also local variable `status` .

Reply:
done